### PR TITLE
df tests and api improved

### PR DESCRIPTION
**Pull request by PoornimaShridhar**

### CODE REVIEW

File: discussion_forum/api.py [hunk 1]
```diff
@@ -88,10 +87,8 @@ def list_discussion_threads(self, request, pk):
         Raises:
             HTTP_401_UNAUTHORIZED: Raised by `IsInstructorOrTAOrStudent`
                 permission class
-            HTTP_403_FORBIDDEN: Raised by `IsInstructorOrTAOrStudent` permission class
-            HTTP_404_NOT_FOUND: Raised by `_is_registered()` method
+            HTTP_403_FORBIDDEN: Raised by `_is_registered()` method
         """
-
         discussion_threads = DiscussionThread.objects.select_related(
```
Comment (suraj-iitb) [inline discussion_forum/api.py]: use get instead of filter with try catch

Comment (suraj-iitb) [reply discussion_forum/api.py]: do till discussion_forum and finally use course_id to get course id

Comment (suraj-iitb) [reply discussion_forum/api.py]: Add exception 404_NOT_FOUND if the thread does not exist,

Comment (suraj-iitb) [reply discussion_forum/api.py]: In docstring also add exception

File: discussion_forum/api.py [hunk 2]
```diff
@@ -151,8 +140,6 @@ def update_discussion_thread(self, request, pk):
             HTTP_401_UNAUTHORIZED: Raised by `IsInstructorOrTAOrStudent`
```
Comment (suraj-iitb) [inline discussion_forum/api.py:145]: 400_BAD_REQUEST is missed in docstring

File: discussion_forum/api.py (hunk 1 above)
Comment (suraj-iitb) [reply discussion_forum/api.py]: add logging also if exception raised

Comment (PoornimaShridhar) [reply discussion_forum/api.py]: done

**Pull request closed without merging**